### ISSUE TITLE
Domained audiences do not assume the same issuer

### ISSUE THREAD

**Issue by candleshine** [sakuraapi/auth-native-authority#42]
the previous domained audience config looked like this 
```
      "jwt": {
      "key": "Zz%TaLM0IM^Ub%I8evp&jdo#cqpBcoKa",
      "issuer": "domained-test-issuer",
      "domainedAudiences": {
        "domain1": {
          "audience1": "132jwS6xavXHKXFwNe2396GmO!6XIF&1",
          "audience2": "232jwS6xavXHKXFwNe2396GmO!6XIF&2",
          "audience3": "332jwS6xavXHKXFwNe2396GmO!6XIF&3"
        },
        "domain2": {
          "audience3": "332jwS6xavXHKXFwNe2396GmO!6XIF&3"
        }
      }
```

[auth-audience](https://github.com/sakuraapi/auth-audience) now looks like this
```
"jwt": {
      "domainedAudiences": {
        "domain1": {
          "audience1": "132jwS6xavXHKXFwNe2396GmO!6XIF&1",
          "audience2": "232jwS6xavXHKXFwNe2396GmO!6XIF&2",
          "audience3": "332jwS6xavXHKXFwNe2396GmO!6XIF&3",
          "issuer": "domained-test-issuer",
          "key": "Zz%TaLM0IM^Ub%I8evp&jdo#cqpBcoKa",
        },
        "domain2": {
          "audience3": "332jwS6xavXHKXFwNe2396GmO!6XIF&3",
          "issuer": "domained-test-issuer",
          "key": "vdEP5D-G_1vb~6i/hA)uVU$X|m{P00[{9"
        }
      }
```

**Comment by candleshine** [comment]
In the scenario above, when a user requests tokens for domain1, the dictionary of tokens will be 4 tokens in length.  
Token 1 will be the reflexive token = `{domain: domain1, iss: domained-test-issuer, aud: domained-test-issuer, key: "Zz%TaLM0IM^Ub%I8evp&jdo#cqpBcoKa"`
Token 2 = `{domain: domain1, iss: domained-test-issuer, aud: audiene1, key: "132jwS6xavXHKXFwNe2396GmO!6XIF&1"`
Token 3 = `{domain: domain1, iss: domained-test-issuer, aud: audiene2, key: "232jwS6xavXHKXFwNe2396GmO!6XIF&2"`
Token 4 = `{domain: domain1, iss: domained-test-issuer, aud: audiene3, key: "332jwS6xavXHKXFwNe2396GmO!6XIF&3"`

**Comment by etsuo** [comment]
released on 0.6.0
